src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2718,21 +2710,22 @@ private static class Holder {
         }
     }
 
-    // package-private
-    ProtectionDomain protectionDomain() {
-        ProtectionDomain pd = getProtectionDomain0();
-        if (pd == null) {
+    /**
+     * Returns the {@code ProtectionDomain} of this class.
+     *
+     * @return the ProtectionDomain of this class
+     *
+     * @see java.security.ProtectionDomain
+     * @since 1.2
+     */
+    public ProtectionDomain getProtectionDomain() {
+        if (protectionDomain == null) {
```

Comment:
Does this imply the class is a primitive class?

Reply:
No I believe classes that are bootstrap classes don't generally have a protection domain.